br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -684,7 +692,20 @@ outloop:
 			return 0.0, false, errors.Trace(err)
 		}
 		lastRow := parser.LastRow()
+		if rowCount == 0 {
+			lastRowLen := len(lastRow.Row)
+			extendColsMap := make(map[string]int)
+			for i, c := range extendCols {
+				extendColsMap[c] = lastRowLen + i
+			}
+			for i, col := range tableInfo.Columns {
```

Comment:
if downstream forget to created extended columns, `lastRow.Row` will has larger length than `columnPermutation`? If so, seems for TiDB backend it will report error "column count mismatch" which is not easy to understand, and for local backend the extended value is ignored which is a waste of time because data is not distinguished.

we might raise a friendly error in this function